kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/ZooKeeperUtils.java

```diff
@@ -26,42 +28,43 @@
  */
 @Slf4j
 public class ZooKeeperUtils {
-    public static void createPath(ZooKeeper zooKeeper, String zkPath, String subPath, byte[] data) {
+
+    public static boolean tryCreatePath(ZooKeeper zooKeeper, String path, byte[] data) {
         try {
-            if (zooKeeper.exists(zkPath, false) == null) {
-                zooKeeper.create(zkPath,
-                        new byte[0], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+            zooKeeper.create(path, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+            if (log.isDebugEnabled()) {
+                log.debug("Created ZK path: {} data: {}", path, new String(data, StandardCharsets.UTF_8));
             }
-            String addSubPath = zkPath + subPath;
-            if (zooKeeper.exists(addSubPath, false) == null) {
-                zooKeeper.create(addSubPath,
-                        data, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
-            } else {
-                zooKeeper.setData(addSubPath, data, -1);
+        } catch (KeeperException e) {
```

Comment:
use 
```
 catch (KeeperException | InterruptedException e ) {
     // ...
 }
```
may better

Reply:
How to check the error code if these two exceptions are merged?

```java
        } catch (KeeperException | InterruptedException e) {
            if (e instanceof KeeperException && ((KeeperException) e).code().equals(KeeperException.Code.NODEEXISTS)) {
                return true;
            } else {
                log.error("Failed to create ZooKeeper node {}: {}", path, e.getMessage());
                return false;
            }
        }
```

IMO, the code logic will be more complicated.

Reply:
OK, just now I found the returned value is ignored. So I won't check the code of KeeperException and merge these exceptions.

Reply:
Done.